HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2341,6 +2346,301 @@ def calc_bounding_values(self):
+            
+        num_pointsM: float
+                Number of gridpoints for market resources grid.
+        
+        num_pointsP: float
+                 Number of gridpoints for permanent income. 
+                 This grid will be exponentiated by the function make_grid_exp_mult.
+                
+        max_p_fac : float
+                Factor that scales the maximum value of permanent income grid. 
+                Larger values increases the maximum value of permanent income grid.
+        
+        Returns
+        -------
+        None
+        '''  
+ 
+   
+        if not hasattr(self, "neutral_measure"):
+            self.neutral_measure = False
+            
+        if num_pointsM == None:
+            m_points = self.aXtraCount
+        else:
+            m_points = num_pointsM
+          
+        if self.cycles == 0:
+            if dist_mGrid == None:    
+                aXtra_Grid = make_grid_exp_mult(
+                        ming=self.aXtraMin, maxg=self.aXtraMax, ng = m_points, timestonest = 3) #Generate Market resources grid given density and number of points
+                
+                for i in range(m_density):
+                    axtra_shifted = np.delete(aXtra_Grid,-1) 
+                    axtra_shifted = np.insert(axtra_shifted, 0,1.00000000e-04)
+                    dist_betw_pts = aXtra_Grid - axtra_shifted
+                    dist_betw_pts_half = dist_betw_pts/2
+                    new_A_grid = axtra_shifted + dist_betw_pts_half
+                    aXtra_Grid = np.concatenate((aXtra_Grid,new_A_grid))
+                    aXtra_Grid = np.sort(aXtra_Grid)
+                self.dist_mGrid =  aXtra_Grid
+                
+            else:
+                self.dist_mGrid = dist_mGrid #If grid of market resources prespecified then use as mgrid
+                
+            if dist_pGrid == None:
+                num_points = num_pointsP #Number of permanent income gridpoints
+                #Dist_pGrid is taken to cover most of the ergodic distribution
+                p_variance = self.PermShkStd[0]**2 #set variance of permanent income shocks
+                max_p = max_p_fac*(p_variance/(1-self.LivPrb[0]))**0.5 #Maximum Permanent income value
+                one_sided_grid = make_grid_exp_mult(1.05+1e-3, np.exp(max_p), num_points, 2)
+                self.dist_pGrid = np.append(np.append(1.0/np.fliplr([one_sided_grid])[0],np.ones(1)),one_sided_grid) #Compute permanent income grid
+                
+            else:
+                self.dist_pGrid = dist_pGrid #If grid of permanent income prespecified then use it as pgrid
+                            
+            if self.neutral_measure == True: # Use Harmenberg 2021's Neutral Measure
+                
+                self.dist_pGrid = np.array([1])
+                
+        elif self.cycles > 1:
+            raise Exception('define_distribution_grid requires cycles = 0 or cycles = 1')
+        
+        elif self.T_cycle != 0:
```

Comment:
So this condition is (cycles == 1 and T_cycle !=0)

Reply:
is `T_cycle < 1` ever?

Reply:
yes this condition is meant as  (cycles == 1 and T_cycle !=0). 

T_cycle is never less than 1 I believe